service/worker/esanalyzer/workflow.go

```diff
@@ -241,57 +247,70 @@ func getFindStuckWorkflowsQuery(
       },
       "size": %d
     }
-    `, startDateTime, endTime, workflowType, maxNumWorkflows)
+    `, startDateTime, endTime, wfTypeMarshaled, maxNumWorkflows), nil
 }
 
-// FindStuckWorkflows is activity to find open workflows that are live significantly longer than average
-func FindStuckWorkflows(ctx context.Context, info WorkflowTypeInfo) (*[]WorkflowInfo, error) {
+// findStuckWorkflows is activity to find open workflows that are live significantly longer than average
+func findStuckWorkflows(ctx context.Context, info WorkflowTypeInfo) ([]WorkflowInfo, error) {
 	analyzer := ctx.Value(analyzerContextKey).(*Analyzer)
+	logger := activity.GetLogger(ctx)
 
 	minNumWorkflowsNeeded := int64(analyzer.config.ESAnalyzerMinNumWorkflowsForAvg(info.Name))
 	if len(analyzer.config.ESAnalyzerLimitToTypes()) > 0 {
-		analyzer.logger.Info("Skipping minimum workflow count validation since workflow types were passed from config")
-	} else if info.NumWorfklows < minNumWorkflowsNeeded {
-		analyzer.logger.Warn(fmt.Sprintf(
+		logger.Info("Skipping minimum workflow count validation since workflow types were passed from config")
+	} else if info.NumWorkflows < minNumWorkflowsNeeded {
+		logger.Warn(fmt.Sprintf(
 			"Skipping workflow type '%s' because it doesn't have enough(%d) workflows to avg",
 			info.Name,
 			minNumWorkflowsNeeded,
 		))
-		return &[]WorkflowInfo{}, nil
+		return nil, nil
 	}
 
-	startDateTime := time.Now().AddDate(0, 0, -analyzer.config.ESAnalyzerLastNDays()).UnixNano()
+	startDateTime := time.Now().Add(-analyzer.config.ESAnalyzerTimeWindow()).UnixNano()
 
 	// allow some buffer time to any workflow
-	maxEndTimeAllowed := time.Now().Add(
-		-time.Second * time.Duration(
-			analyzer.config.ESAnalyzerBufferWaitTimeInSeconds(info.Name))).UnixNano()
+	maxEndTimeAllowed := time.Now().Add(-analyzer.config.ESAnalyzerBufferWaitTime(info.Name)).UnixNano()
+
 	// if the workflow exec time takes longer than 3x avg time, we refresh
 	endTime := time.Now().Add(
-		-time.Second * time.Duration((int64(info.Duration.AvgExecTime) * 3)),
+		-time.Nanosecond * time.Duration((int64(info.Duration.AvgExecTimeNanoseconds) * 3)),
```

Comment:
I think you can get rid of the `-time.Nanosecond` here, as it's already a duration, plus it's literally equal to 1: https://pkg.go.dev/time#pkg-constants